code/controllers/subsystem/input.dm

```diff
@@ -115,6 +115,8 @@ SUBSYSTEM_DEF(input)
 
 /datum/controller/subsystem/input/fire()
 	var/list/clients = GLOB.clients // Let's sing the list cache song
+	if(listclearnulls(clients)) // clear nulls before we run keyloop
+		log_world("Found a null in clients list!")
 	for(var/i in 1 to clients.len)
```

Comment:
Why not `for var/client/C in clients`?

Reply:
Checking if each client is null seems more demanding than checking if there is a null in the global client list before running through the list

Reply:
Oh wait I see what you mean now